### PR TITLE
qe7.3.1 for gh200

### PR DESCRIPTION
**Pull request by simonpintarelli**
TODO: libxc required to disable patch for `configure` (which is generated by autoconf, but missing at the time it wants to apply the patch) -> open an issue in spack upstream

### CODE REVIEW

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantum_espresso:7.3.1

File: recipes/quantumespresso/v7.3.1/gh200/modules.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    tcl:
+      all:
+        autoload: none
+      hash_length: 0
+      exclude_implicits: true
+      exclude: ['%gcc@7.5.0', 'gcc %gcc@7.5.0']
+      projections:
+        all: '{name}/{version}'
+        #cray-mpich-binary: 'cray-mpich-{compiler.name}'
+        #libxc: '{name}/{version}-{compiler.name}'
```
Comment (RMeli) [inline recipes/quantumespresso/v7.3.1/gh200/modules.yaml:25]:
```suggestion
```
if not needed

**Comment by RMeli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantum_espresso:7.3.1

**Review by RMeli — COMMENTED** [review]
This new configuration needs to be added to `config.yaml`.

**Comment by RMeli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantum_espresso:7.3.1

**Comment by RMeli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantumespresso:7.3.1

**Comment by RMeli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantumespresso:v7.3.1

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantumespresso:v7.3.1

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantumespresso:v7.3.1

File: recipes/quantumespresso/v7.3.1/gh200/repo/packages/nvpl-blas/package.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
```
Comment (bcumming) [inline recipes/quantumespresso/v7.3.1/gh200/repo/packages/nvpl-blas/package.py:1]: Should we move the `nvpl-blas` and `nvpl-lapack` to the site repo, for all uenv on GH200 to use?

File: recipes/quantumespresso/v7.3.1/gh200/environments.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  - patchelf%gcc
+  - git%gcc
+  variants:
+  - +mpi
+  - +cuda
+  - cuda_arch=90
+  views:
+    default:
+    develop:
+      exclude: ['quantum-espresso']
```
Comment (bcumming) [inline recipes/quantumespresso/v7.3.1/gh200/environments.yaml:27]: Should these views also exclude `git` and `patchelf`?

Comment (bcumming) [reply recipes/quantumespresso/v7.3.1/gh200/environments.yaml:27]: You could also try adding `git` and `patchelf` as
```yaml
  packages:
    - git
    - patchelf
```

**Review by bcumming — CHANGES_REQUESTED** [review]
Thanks!

I have just got a few small nit-picks.

File: recipes/quantumespresso/v7.3.1/gh200/environments.yaml (hunk 3 above)
Comment (simonpintarelli) [reply recipes/quantumespresso/v7.3.1/gh200/environments.yaml:27]: there is no patchelf installed on santis.

I will add `git` to packages.

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantumespresso:v7.3.1

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantumespresso:v7.3.1

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantumespresso:v7.3.1

**Comment by bcumming** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=quantumespresso:v7.3.1

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantumespresso:v7.3.1

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=quantumespresso:v7.3.1

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantumespresso:v7.3.1

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=quantumespresso:v7.3.1

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=santis;uarch=gh200;uenv=quantumespresso:v7.3.1

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=quantumespresso:v7.3.1

**Comment by simonpintarelli** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=quantumespresso:v7.3.1

**Review by bcumming — APPROVED** [review]

**Pull request merged**